### PR TITLE
docs: run unit tests for docs app in CI

### LINKED ISSUE #1626: Docs: run unit tests for docs app
We aren't currently running the suite of unit tests for the docs app because of changes made after the initial port. 

This effort would:

- [ ] Run existing suite
- [ ] Fix necessary test and disable unnecessary ones
- [ ] Re-enable the docs tests in CI

### PR DESCRIPTION
**Pull request by jtcrowson**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

ngrx.io tests do not run in CI and fail when running locally

Closes #1626 

## What is the new behavior?

ngrx.io tests successfully run and pass in CI

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jtcrowson** [comment]
@brandonroberts ready for review.  The AppComponent tests were almost completely broken, so I left the ones that worked and rewrote the rest.

File: .circleci/config.yml [hunk 1]
```diff
@@ -117,6 +117,16 @@ jobs:
       - run: yarn run example:build:prod
       - run: yarn run example:cypress:ci
 
+  docs-tests:
+    docker:
+      - image: circleci/node:10.12-browsers
+    steps:
+      - checkout
+      - restore_cache:
+          key: *cache_key
```
Comment (brandonroberts) [inline .circleci/config.yml]:
```suggestion
          key: *docs_cache_key
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -117,6 +117,16 @@ jobs:
       - run: yarn run example:build:prod
       - run: yarn run example:cypress:ci
 
+  docs-tests:
+    docker:
```
Comment (brandonroberts) [inline .circleci/config.yml]:
```suggestion
    <<: *run_in_node
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -117,6 +117,16 @@ jobs:
...
+      - image: circleci/node:10.12-browsers
+    steps:
+      - checkout
+      - restore_cache:
+          key: *cache_key
+      - run: yarn --cwd projects/ngrx.io/ setup
```
Comment (brandonroberts) [inline .circleci/config.yml]: I don't think we need this step if we restore the docs cache

File: .circleci/config.yml [hunk 4]
```diff
@@ -117,6 +117,14 @@ jobs:
       - run: yarn run example:build:prod
       - run: yarn run example:cypress:ci
 
+  docs-tests:
```
Comment (brandonroberts) [inline .circleci/config.yml]: I believe the initial setup should match lines 129-135 so the cache is restored correctly

**Comment by jtcrowson** [comment]
@brandonroberts made those change and now it's erroring on `No binary for Chrome browser on your platform`.

**Comment by brandonroberts** [comment]
@jtcrowson My fault. I forgot those tests aren't running in node. Will you create a `run_in_browser` entry and use the images you had previously?

**Comment by jtcrowson** [comment]
@brandonroberts reverted back to previous config and create a run_in_browser entry in 734edd3.  Then tried to restore cache using `docs_cache_key` and removing the `yarn setup` step in abbc273, but the build complained that  `yarn setup` needed to be run.  So I've reverted abbc273 back to the previous working version in b2026b6.  Let me know if there's anything else you'd like me to try.

**Comment by brandonroberts** [comment]
@jtcrowson try this. If that doesn't work, using the `yarn setup` will be fine.

```
  docs-tests:
    <<: *run_in_browser
    working_directory: ~/docs/projects/ngrx.io
    steps:
      - checkout:
          path: ~/docs
      - restore_cache:
          key: *docs_cache_key
      - run: yarn --cwd projects/ngrx.io/ test --watch=false
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -99,24 +101,32 @@ jobs:
...
+    <<: *run_in_browser
+    working_directory: ~/docs/projects/ngrx.io
+    steps:
+      - checkout:
+          path: ~/docs
+      - restore_cache:
+          key: *docs_cache_key
+      - run: yarn --cwd projects/ngrx.io/ test --watch=false
```
Comment (brandonroberts) [inline .circleci/config.yml]:
```suggestion
      - run: yarn test --watch=false
```

Comment (jtcrowson) [reply .circleci/config.yml]: Fixed in 19c1e2a

**Comment by jtcrowson** [comment]
@brandonroberts thanks for your help! 💯

**Pull request merged**